client/internal/httplooper.go

```diff
@@ -0,0 +1,256 @@
+package internal
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"sync/atomic"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+	"github.com/open-telemetry/opamp-go/internal"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/open-telemetry/opamp-go/client/types"
+	"github.com/open-telemetry/opamp-go/protobufs"
+)
+
+const OpAMPPlainHTTPMethod = "POST"
+
+// HTTPLooper allows scheduling messages to send. Once run, it will loop through
+// a request/response cycle for each message to send and will process all received
+// responses using a receivedProcessor. If there are no pending messages to send
+// the HTTPLooper will wait for the configured polling interval.
+type HTTPLooper struct {
+	url               string
+	instanceUid       atomic.Value
+	logger            types.Logger
+	client            *http.Client
+	callbacks         types.Callbacks
+	pollingIntervalMs int64
+
+	// Headers to send with all requests.
+	requestHeader http.Header
+
+	// Indicates that there is a pending message to send.
+	hasPendingMessage chan struct{}
+
+	// The next message to send.
+	nextMessage nextMessage
+
+	// Processor to handle received messages.
+	receiveProcessor receivedProcessor
+}
+
+func NewHTTPLooper(logger types.Logger) *HTTPLooper {
+	h := &HTTPLooper{
+		logger:            logger,
+		client:            http.DefaultClient,
+		hasPendingMessage: make(chan struct{}, 1),
+		requestHeader:     http.Header{},
+		pollingIntervalMs: 30 * 1000, // default interval is 30 seconds.
```

Comment:
This is a nit but I would expose the default as a constant in code

Reply:
Fixed.